src/analysis/special_functions/trigonometric.lean

```diff
@@ -1748,15 +1750,111 @@ by simp only [tan_eq_sin_div_cos]; exact
+begin
+  refine continuous_on_tan.mono _,
+  intros x hx,
+  simp only [mem_set_of_eq],
+  exact ne_of_gt (cos_pos_of_mem_Ioo hx.1 hx.2),
+end
+
+open filter
+open_locale topological_space
+
+lemma tendsto_sin_pi_div_two : tendsto sin (𝓝[Iio (π/2)] (π/2)) (𝓝 1) :=
+by { convert continuous_sin.continuous_within_at, simp }
+
+lemma tendsto_cos_pi_div_two : tendsto cos (𝓝[Iio (π/2)] (π/2)) (𝓝[Ioi 0] 0) :=
+begin
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within,
+  { convert continuous_cos.continuous_within_at, simp },
+  { filter_upwards [Ioo_mem_nhds_within_Iio (right_mem_Ioc.mpr (norm_num.lt_neg_pos
+      _ _ pi_div_two_pos pi_div_two_pos))] λ x hx, cos_pos_of_mem_Ioo hx.1 hx.2 },
+end
+
+lemma tendsto_tan_pi_div_two : tendsto tan (𝓝[Iio (π/2)] (π/2)) at_top :=
+begin
+  convert tendsto_mul_at_top (by norm_num) (tendsto.inv_tendsto_zero tendsto_cos_pi_div_two)
+            tendsto_sin_pi_div_two,
+  ext x,
+  rw tan_eq_sin_div_cos x,
+  ring,
+end
+
+lemma tendsto_sin_neg_pi_div_two : tendsto sin (𝓝[Ioi (-(π/2))] (-(π/2))) (𝓝 (-1)) :=
+by { convert continuous_sin.continuous_within_at, simp }
+
+lemma tendsto_cos_neg_pi_div_two : tendsto cos (𝓝[Ioi (-(π/2))] (-(π/2))) (𝓝[Ioi 0] 0) :=
+begin
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within,
+  { convert continuous_cos.continuous_within_at, simp },
+  { filter_upwards [Ioo_mem_nhds_within_Ioi (set.left_mem_Ico.mpr (norm_num.lt_neg_pos
+      _ _ pi_div_two_pos pi_div_two_pos))] λ x hx, cos_pos_of_mem_Ioo hx.1 hx.2 },
+end
+
+lemma tendsto_tan_neg_pi_div_two : tendsto tan (𝓝[Ioi (-(π/2))] (-(π/2))) at_bot :=
+begin
+  convert tendsto_mul_at_bot (by norm_num) (tendsto.inv_tendsto_zero tendsto_cos_neg_pi_div_two)
+            tendsto_sin_neg_pi_div_two,
+  ext x,
+  rw tan_eq_sin_div_cos x,
+  ring,
+end
+
+/-!
+### Continuity and differentiability of arctan
+
+The continuity of `arctan` is difficult to prove due to `arctan` being (indirectly) defined naively
+via `classical.some`. The proof therefore uses the general theorem that monotone functions are
+homeomorphisms: `homeomorph_of_strict_mono_continuous_Ioo`. We first prove that `tan` (restricted)
+is a homeomorphism, then that `arctan` is equal (though not defeq) to the inverse function of that
+homeomorphism. The fact that `arctan` is continuous is then derived from the fact that it is equal
+to a homeomorphism, and its differentiability is in turn derived from its continuity using
+`has_deriv_at.of_local_left_inverse`.
+-/
+
+/-- The function `tan`, restricted to the open interval (-π/2, π/2), is a homeomorphism -/
+def tan_homeomorph : homeomorph (Ioo (-(π/2)) (π/2)) ℝ :=
+homeomorph_of_strict_mono_continuous_Ioo tan (by linarith [pi_div_two_pos])
+  (λ x y, tan_lt_tan_of_lt_of_lt_pi_div_two) continuous_on_tan_of_mem_Ioo tendsto_tan_pi_div_two
+    tendsto_tan_neg_pi_div_two
+
+lemma arctan_eq_tan_homeomorph_inv_fun : arctan = coe ∘ tan_homeomorph.inv_fun :=
```

Comment:
It would probably be more convenient if `\la x, ⟨arctan x, arctan_mem_Ioo x⟩` is the inverse of `tan_homeomorph` by definition. We could do this by having a definition that takes `f : A ≃ₜ B` and a right-inverse `g : B -> A` of `f`, and spits out a new homeomorphism `A ≃ₜ B` with inverse definitionally equal to `g`.

This lemma is probably even useful in that case (for rewriting), but the order should be reversed (currently the LHS is simpler), and hence the name should be changed.

Reply:
> It would probably be more convenient if `\la x, ⟨arctan x, arctan_mem_Ioo x⟩` is the inverse of `tan_homeomorph` by definition.

An alternative way to achieve this: change the definition of `arctan`, so that it is equal to the (coerced) inverse of this homeomorphism by definition.  The [current definition](https://leanprover-community.github.io/mathlib_docs/analysis/special_functions/trigonometric.html#real.arcsin) in terms of `arcsin` could then become a lemma. (@fpvandoorn?)

Reply:
Yes, another option is to redefine `arctan`, but that probably requires more changes in other parts.

Reply:
But if the first thing proved after the definition is a lemma giving the old definition, the change shouldn't be too bad?  Maybe I underestimate it :)